api/v2/submitblock.go

```diff
@@ -0,0 +1,29 @@
+package v2
+
+import (
+	"encoding/json"
+	"fmt"
+
+	v1 "github.com/attestantio/go-builder-client/api/v1"
+	"github.com/attestantio/go-eth2-client/spec/bellatrix"
+	"github.com/attestantio/go-eth2-client/spec/capella"
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+)
+
+// SubmitBlockRequest is the v2 request from the builder to submit a block.
+type SubmitBlockRequest struct {
+	Message                *v1.BidTrace
+	ExecutionPayloadHeader *capella.ExecutionPayloadHeader
+	Signature              phase0.BLSSignature `ssz-size:"96"`
+	Transactions           []bellatrix.Transaction
+	Withdrawals            []*capella.Withdrawal
```

Comment:
may be splitting hairs but again if we really want to think about perf then why have the pointer indirection on `Withdrawals`? not exactly sure why this pattern is in `go-eth2-client`

Reply:
Do you think the pointer indirection negatively impacts performance?

Fwiw, I don't think it's needed for the withdrawals. It is commonly used for headers and payloads in this project because the type allows both bellatrix and capella types to be embedded, with the unused just being nil.

Reply:
ya the over the wire bytes will be the exact same, its just about the marshal and unmarshal into the struct. no strong opinion here, but agree that the performance implications are probably negligible. from the relay perspective, once it has the `SubmitBlockRequest` message in memory, it is passing around a pointer to that object anyway, and any copies have the full `[]Transaction` slice which is a few order of magnitudes bigger than the withdrawals. 

the main reason i went with the pointer was to match the style. would love to hear if @mcdee has an opinion on this. if not, i would lean towards a value slice, which is only a two line change to this PR :-)

Reply:
The general style is to have pointers to structs; the cost of dereferencing is minimal in the grand scheme of things, and for structs that have variable elements it allows for easier memory management (`[]*AStruct` can be easily allocated with a fixed size capacity and not have to worry about reallocations).

Reply:
In that way, wouldn't we also want to use pointers for the transactions?

Reply:
`Transaction` is typed to `[]byte`, so isn't a struct in itself (and `[]byte` is a fixed-size element so again no need to worry about reallocations).